### ISSUE TITLE
Script surrogates?

### ISSUE THREAD

**Issue by pde** [EFForg/privacybadgerchrome#400]
It seems that NoScript and Ghostery have worked around a lot of script-blocking breakage by writing drop in replacement "surrogate" scripts for common third parties.  The NoScript version is open source, and mirrored on GitHub [here](https://github.com/avian2/noscript/blob/master/xpi/defaults/preferences/noscript.js).

We should consider whether we can transpose some or all of this set of scripts straight into socialwidgets.json (or a separate semantically equivalent file); it might save us a lot of breakage with the extra non-iframed script blocking we're going to be doing, and also allow us to shrink the yellowlist.

**Comment by pde** [comment]
Why don't we try to just copying some of the surrogates into `socialwidgets.json`, and seeing if they work?

**Comment by cooperq** [comment]
that sounds like it's worth a shot.

**Comment by TMarkGibson** [comment]
It would be wonderful it cloudfront.net was yellowlisted.  I have no idea about actual ads which are associated with cloudfront.net but I do know that it is functioning as a tracker.  There are certain sites which fail to function if one  blocks all URLs of the form https://nnnnnnnnnnnnnnn.cloudfront.net where n can be any single digit or lowercase letter.

These fourteen character lower-case alphanumeric strings are [pseudo]randomly generated and prepended to ".cloudfront.net" so that it is not possible to block just a few cloudfront.net trackers unless they are on the yellowlist so that the vast majority can be tossed into the red category (reaD: blacklist) by default.

At present, I use one site frequently that I allow less than a dozen cloudfront.net URLs generated by said site to function with the rest blocked by NoScript and/or uBlock Origin.  I tend to use both, along with Privacy Badger and the free version of Ghostery.

The interesting thing is that on sites that use these random cloudfront.net URLs, one can always block ones that are used by a particular site and it (in my experience) will generate however many new ones it needs when you browse that site again, apparently as a first-time visitor.

So, if one wants to make it more difficult for cloudfront.net to be used for tracking purposes, it is easy enough to swap in one set of active randomly generated cloudfront.net URLs and just use a few different sets so that you are recognized as a unique user per set on that site.

The problem with that it that one also has to make sure the computer has a constantly changing finger print, and some elements of a typical fingerprint are harder for a user to control than others, so the computer will likely fall into a bin of machine fingerprints that have the hard to control elements in common.

If you look at who is behind cloudfront.net, you can understand why you might want to use the randomized URLs you currently allow on one site when you visit another site for which you maintain some persistent randomly generated URLs.

By using VMs (which I do not) one could make it much harder for cloudfront.net to be used as a tracker of a particular platform, but raises flags if you are seen to be connecting from several locations around the world with a system that shows signs of being located in a particular apartment complex in a U.S. city.  I have not used VMs enough to understand just how one would go about visiting the same site several times a week, 52/365 without attracting attention of whoever is using cloudront.net as a tracker on a few sites you use often.

If it were up to me, cloudfront.net would be Redlisted by default, but Yellowlisting a lot less drastic and does not seem to break things on any site that I frequent and allow a set of cloudfront.net URLs
it recognizes from some of my previous visits.

There is a point to this long screed: if I want to use some sites that utilize cloudfront.net, all the (partially) blocked cloudfront.net will pile up as entries in NoScript and uBlock Origin or UMatrix.  Having cloudfront.net Yellowlisted by default would let me change that default to Redlist except in a select few cases, thereby eliminating the need for me to keep cleaning out the cloudfront trash from the <<2K listed sites being blocked by my computer using various software tools.

In essence, cloudfront.net will always create a lot of unwanted entries in Privacy Badger, uBlock Origin and uMatrix (which works well in conjunction with the other blocker written by the same clever, hardworking and diligent person.

PLEASE, Yellowlist cloudfront.net so I can block it by changing the default to Redlist yet non-technical users will not have problems with it being on the Yellowlist (judging from my experiences).
This is a case where the URL itself will function as an tracker but can be more easily blocked without cluttering up tables in blocking software.

I need a nap.
Regards,
Mark Gibson

**Comment by ghostwords** [comment]
[Cloudfront is already on the yellowlist](https://github.com/EFForg/privacybadger/blob/4c3f4c7eb38904a8b300765f8f585290e35dad4c/src/data/yellowlist.txt#L99). If you'd like to continue this discussion, could you open a new issue instead of posting on this unrelated issue?

**Comment by TMarkGibson** [comment]
I got a very quick reply in email saying that my request to have cloudfront.net added to the yellowlist was off-topic since cloudfront.net is already there.

I looked at the link to the yellowlist he provided and sure enough, it was there, just the way I asked for in the poorly researched request I made.

Now I wonder why that list is not updated on my computers, because it has not been present when I wanted to set the default to redlist for cloudflare.net and then allow those few randomly generated 3rd level parts of the URL that I wanted to allow as exceptions.  I guess I ought to start by looking to see how old my installation of PrivacyBadger is and find out why it is not current if that is the case.

I really do not want to lose all the work I have put into tweaking the PrivacyBadger settings to make it work extremely well with other blocking software I use.  Once I get an updated yellowlist installed, it will make life a lot easier for me, as cloudfront.net has been a major peeve of mine as far as how my installation fails to have an entry for it by default.  Was it added fairly recently?

Otherwise, I think PrivacyBadger is great because it is speedy and removes the need for me to adjust other blockers as often as I have had to before I got it.

Is there someplace I might find detailed instructions on keeping PrivacyBadger current without stepping on my settings and alterations to the potential tracker list?  Looking at the data management tab in the add-on, I see that what I want to do would likely involve some Cloud -- I tend to avoid using Cloud storage for security reasons and the potential lack of control of my own data if the cloud provider gets wonky -- which is something I would have to set up specifically for PrivacyBadger (not a real problem).

Regards,
Mark
